src/oc/web/components/ui/wrt.cljs

```diff
@@ -0,0 +1,162 @@
+(ns oc.web.components.ui.wrt
+  (:require [rum.core :as rum]
+            [cuerdas.core :as string]
+            [org.martinklepsch.derivatives :as drv]
+            [oc.web.lib.utils :as utils]
+            [oc.web.lib.responsive :as responsive]
+            [oc.web.actions.activity :as activity-actions]
+            [oc.web.actions.notifications :as notifications-actions]
+            [oc.web.components.ui.user-avatar :refer (user-avatar-image)]))
+
+(defn filter-by-query [user query]
+  (or (string/includes? (string/lower (:email user)) query)
+      (string/includes? (string/lower (:name user)) query)
+      (string/includes? (string/lower (:first-name user)) query)
+      (string/includes? (string/lower (:last-name user)) query)))
+
+(defn under-middle-screen? [el]
+  (let [el-offset-top (aget (.offset (js/$ el)) "top")
+        fixed-top-position (- el-offset-top (.-scrollTop (.-scrollingElement js/document)))
+        win-height (.-innerHeight js/window)]
+    (>= fixed-top-position (/ win-height 2))))
+
+(defn calc-left-position [el]
+  (let [el-offset-left (aget (.offset (js/$ el)) "left")
+        win-width (.-innerWidth js/window)]
+    (if (> (+ el-offset-left 360) (- win-width 40))
+      (- (- win-width 40) (+ el-offset-left 360))
+      0)))
+
+(def default-appear-delay 30)
+(def default-disappear-delay 500)
+
+(rum/defcs wrt < rum/reactive
+                 ;; Derivatives
+                 (drv/drv :team-roster)
+                 ;; Locals
+                 (rum/local "" ::query)
+                 (rum/local false ::showing-popup)
+                 (rum/local false ::under-middle-screen)
+                 (rum/local 0 ::left-position)
+                 (rum/local :seen ::list-view)
+                 (rum/local nil ::show-delay)
+
+                 {:after-render (fn [s] (.tooltip (js/$ "[data-toggle=\"tooltip\"]")) s)}
+
+  [s activity-data read-data]
+  (let [item-id (:uuid activity-data)
+        team-roster (drv/react s :team-roster)
+        team-users (filter #(= (:status %) "active") (:users team-roster))
+        seen-users (into [] (sort-by utils/name-or-email (map #(assoc % :seen true) (:reads read-data))))
```

Comment:
This seen data belongs in the store/app-state instead of here in the component. You can transform all of this data and make it available in the app state. You can add it to the `:activities-read` data.  Something like `:activities-data [{uuid {:seen-users [], :unseen-users[]}}]`.